zerver/lib/bot_lib.py

```diff
@@ -83,7 +89,20 @@ def identity(self) -> BotIdentity:
         return BotIdentity(self.full_name, self.email)
 
     def react(self, message: Dict[str, Any], emoji_name: str) -> Dict[str, Any]:
-        return {}  # Not implemented
+        if "id" not in message:
+            raise EmbeddedBotKeyNotFound(_('Expected to find the "id" required key.'))
+
+        emoji_code, reaction_type = emoji_name_to_emoji_code(self.user_profile.realm, emoji_name)
+
+        do_add_reaction(
+            self.user_profile,
+            Message.objects.filter(id=int(message["id"]))[0],
+            emoji_name,
+            emoji_code,
+            reaction_type,
+        )
+
+        return {}
```

Comment:
Is the right interface for this to return the details of the created reaction in the dictionary?  I can imagine a client wanting those to e.g. unreact naturally.

(Can be left for future work, since that's a coordinated change; we probably also want a `remove_reaction` thing eventually -- just thinking through the direction here).

Reply:
It should look like this,
```json
{
    "msg": "",
    "result": "success"
}
```
I will fix that.

Reply:
Yeah, `remove_reaction` can be added to our protocol in python-zulip-api.

Reply:
I have created an issue for this https://github.com/zulip/python-zulip-api/issues/684 in the Zulip API repo.